pkg/server/plugin/upstreamauthority/vault/vault_test.go

```diff
@@ -112,426 +106,586 @@ func (vps *VaultPluginSuite) Test_Configure() {
+				fakeVaultServer.SignIntermediateReqEndpoint = "/v1/test-pki/root/sign-intermediate"
+				fakeVaultServer.SignIntermediateResponseCode = 200
+				fakeVaultServer.SignIntermediateResponse = []byte(testInvalidSignIntermediateResponse)
+
+				return fakeVaultServer
+			},
+			expectCode:      codes.Internal,
+			expectMsgPrefix: "upstreamauthority(vault): failed to parse Root CA certificate:",
+		},
+		{
+			test: "Signing response malformed certificate",
+			csr:  csr.Raw,
+			fakeServer: func() *FakeVaultServerConfig {
+				fakeVaultServer := setupSuccessFakeVaultServer()
+				// Expect error
+				fakeVaultServer.SignIntermediateReqEndpoint = "/v1/test-pki/root/sign-intermediate"
+				fakeVaultServer.SignIntermediateResponseCode = 200
+				fakeVaultServer.SignIntermediateResponse = []byte(testSignMalformedCertificateResponse)
+
+				return fakeVaultServer
+			},
+			expectCode:      codes.Internal,
+			expectMsgPrefix: "upstreamauthority(vault): failed to parse certificate: no PEM blocks",
+		},
+		{
+			test:            "Invalid CSR",
+			csr:             []byte("malformed-csr"),
+			fakeServer:      setupSuccessFakeVaultServer,
+			expectCode:      codes.InvalidArgument,
+			expectMsgPrefix: "upstreamauthority(vault): failed to parse CSR data:",
+		},
+	} {
+		tt := tt
+		t.Run(tt.test, func(t *testing.T) {
+			fakeVaultServer := tt.fakeServer()
+			s, addr, err := fakeVaultServer.NewTLSServer()
+			require.NoError(t, err)
 
-	v0 := new(upstreamauthority.V0)
-	plugintest.Load(vps.T(), builtin(p), v0)
-	vps.plugin = v0.UpstreamAuthorityClient
+			s.Start()
+			defer s.Close()
 
-	req := vps.loadMintX509CARequestFromTestFile()
+			p := New()
+
+			v1 := new(upstreamauthority.V1)
+			plugintest.Load(t, builtin(p), v1,
+				plugintest.ConfigureJSON(&Configuration{
+					VaultAddr:     fmt.Sprintf("https://%v/", addr),
+					CACertPath:    testRootCert,
+					PKIMountPoint: "test-pki",
+					TokenAuth: &TokenAuthConfig{
+						Token: "test-token",
+					},
+				}),
+				plugintest.CoreConfig(catalog.CoreConfig{TrustDomain: spiffeid.RequireTrustDomainFromString("example.org")}),
+			)
+
+			if tt.configFails {
+				p.cc = nil
+			}
 
-	_, err = vps.mintX509CA(req)
-	vps.Require().Error(err)
+			x509CA, x509Authorities, stream, err := v1.MintX509CA(context.Background(), tt.csr, time.Minute)
+			spiretest.RequireGRPCStatusHasPrefix(t, err, tt.expectCode, tt.expectMsgPrefix)
+			assert.Nil(t, x509CA)
+			assert.Nil(t, x509Authorities)
+			assert.Nil(t, stream)
+		})
+	}
 }
 
-func (vps *VaultPluginSuite) Test_MintX509CA_InvalidVaultResponse() {
-	vps.fakeVaultServer.LookupSelfResponse = []byte(testLookupSelfResponse)
-	vps.fakeVaultServer.LookupSelfResponseCode = 200
-	vps.fakeVaultServer.SignIntermediateReqEndpoint = "/v1/test-pki/root/sign-intermediate"
-	vps.fakeVaultServer.SignIntermediateResponseCode = 200
-	vps.fakeVaultServer.SignIntermediateResponse = []byte(testInvalidSignIntermediateResponse)
+func TestMintX509CA_InvalidCSR(t *testing.T) {
```

Comment:
Why is this not part of the TestMintX509CAFails test above?

Reply:
refactored now to be into the same unit test